src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -59,15 +59,11 @@ public void clear() {
 
     public LinkedList<Product> removeThisProduct(String name) {
         LinkedList<Product> removedProducts = new LinkedList<>();
```

Comment:
```suggestion
        List<Product> removedProducts = new LinkedList<>();
```
не забываем про полиморфизм